src/dom/elements/html-input-element.ts

```diff
@@ -0,0 +1,34 @@
+import type { Node } from "../node.ts";
+import type { CTOR_KEY } from "../../constructor-lock.ts";
+import { Element } from "../element.ts";
+
+export class HTMLInputElement extends Element {
+  constructor(
+    parentNode: Node | null,
+    attributes: [string, string][],
+    key: typeof CTOR_KEY,
+  ) {
+    super(
+      "INPUT",
+      parentNode,
+      attributes,
+      key,
+    );
+  }
+
+  get value() {
+    return this.getAttribute("value") ?? "";
+  }
+  set value(value: any) {
+    this.setAttribute("value", value);
+    this.dispatchEvent(new Event("input"));
```

Comment:
Programmatically setting the values does not actually trigger DOM events. You can check the [very last two notes](https://html.spec.whatwg.org/multipage/input.html#common-input-element-events) in the input element eventing behaviour section of the standard.